docs/session-data.md

```diff
@@ -0,0 +1,160 @@
+
+`onFileData`: [FileSuccessHandler](#FileSuccessHandler)
+
+This callback function is triggered whenever we have received data from the server. The content is passed back in files, so this function will be triggered whenever we have received data from one file. The callback will be triggered with an object in the parameter. The object has the following properties:
+
+`onFileError`: [FileErrorHandler](#FileErrorHandler)
+
+This callback function is triggered whenever we have failed to receive any data for the data file. By default, we try to fetch the data five times with exponential backoff before invoking the error callback. You can configure the retry options when you initiate the SDK. 
+
+#### Exceptions
+[ParameterValidationError](./handling-errors.md)
+
+[FileDecryptionError](./handling-errors.md)
+
+
+## GetSessionDataResponse
+This is what you'll receive when you call `getSessionData()`
+```typescript
+interface GetSessionDataResponse = {
+    stopPolling: () => void;
+    filePromise: Promise;
+}
+```
+
+`stopPolling`: function
+
+A function which stops the polling when you've called `getSessionData()`. This is used to prematurely stop the data fetching before all the user data has been received.
+
+`filePromise`: Promise
+
+A promise that will resolve once all user files have been processed.
+
+
+## FileSuccessHandler
+This is the callback that is triggered whenever we've received data from a user. This function will be called once everytime we've fetched a user file. This may be called multiple times per file if the file has been updated since last time we fetched.
+```typescript
+type FileSuccessHandler = ({
+    fileData: any,
+    fileDescriptor: FileDescriptor,
+    fileName: string,
+    fileList: string[],
+}): void;
+```
+`fileData`: string
+
+JSON string of data objects
+
+`fileDescriptor`: [FileDescriptor](#FileDescriptor)
+
+Object describing data that is returned
+
+`fileName`: string
+
+The filename which the objects reside in
+
+`fileList`: string[]
+
+The list of all files that are to be returned
+
+
+## FileDescriptor
+```typescript
+interface FileDescriptor {
+    objectCount: number;
+    objectType: string;
+    serviceGroup: string;
+    serviceName: string;
+}
+```
+
+`objectCount`: number
+
+How many data objects are returned in this file.
+
+`objectType`: string
+
+What data these objects represent. E.g. Media
+For a full list, please check out our developer docs [here](http://developers.digi.me/reference-objects).
+
+`serviceGroup`: string
+
+What service group the data belongs to. E.g. Social.
+For a full list, please check out our developer docs [here](http://developers.digi.me/reference-objects).
+
+`serviceName`: string
+
+What service the data came from. E.g. Facebook.
+For a full list, please check out our developer docs [here](http://developers.digi.me/reference-objects).
```

Comment:
Are we able to change the link to display `Reference Objects` instead of `here`

Perhaps re-word to `For a full list, please check out our Reference Objects guide on our developer docs.`